src/main/java/com/bazel_diff/BazelSourceFileTarget.java

```diff
@@ -17,51 +16,40 @@ interface BazelSourceFileTarget {
 class BazelSourceFileTargetImpl implements BazelSourceFileTarget {
     private String name;
     private byte[] digest;
+    static private ByteBufferPool pool = new ByteBufferPool(1024, 10240); //10kb
 
-    private void digestLargeFile(MessageDigest finalDigest, FileChannel inChannel) throws IOException {
-        int bufferSize = 10240; // 10kb
-        ByteBuffer buffer = ByteBuffer.allocate(bufferSize);
-        while (inChannel.read(buffer) != -1) {
+    private void digest(Hasher finalDigest, InputStream stream) throws Exception {
+        ByteBuffer buffer = pool.borrow();
+        byte[] array = buffer.array(); //Available for non-direct buffers
```

Comment:
There is an assumption here that the buffer pool does not provide direct memory-based buffers